server/src/test/java/org/apache/druid/server/coordinator/rules/BroadcastDistributionRuleTest.java

```diff
@@ -21,293 +21,214 @@
 
 import org.apache.druid.client.DruidServer;
 import org.apache.druid.java.util.common.DateTimes;
-import org.apache.druid.java.util.common.Intervals;
+import org.apache.druid.java.util.common.granularity.Granularities;
 import org.apache.druid.server.coordination.ServerType;
+import org.apache.druid.server.coordinator.CreateDataSegments;
 import org.apache.druid.server.coordinator.DruidCluster;
 import org.apache.druid.server.coordinator.DruidCoordinatorRuntimeParams;
 import org.apache.druid.server.coordinator.ServerHolder;
 import org.apache.druid.server.coordinator.balancer.RandomBalancerStrategy;
+import org.apache.druid.server.coordinator.loading.SegmentAction;
 import org.apache.druid.server.coordinator.loading.SegmentLoadQueueManager;
 import org.apache.druid.server.coordinator.loading.StrategicSegmentAssigner;
 import org.apache.druid.server.coordinator.loading.TestLoadQueuePeon;
 import org.apache.druid.server.coordinator.stats.CoordinatorRunStats;
+import org.apache.druid.server.coordinator.stats.Dimension;
+import org.apache.druid.server.coordinator.stats.RowKey;
 import org.apache.druid.server.coordinator.stats.Stats;
 import org.apache.druid.timeline.DataSegment;
-import org.apache.druid.timeline.partition.NoneShardSpec;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 
-import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 
 public class BroadcastDistributionRuleTest
 {
-  private DruidCluster druidCluster;
-  private ServerHolder holderOfSmallSegment;
-  private final List<ServerHolder> holdersOfLargeSegments = new ArrayList<>();
-  private final List<ServerHolder> holdersOfLargeSegments2 = new ArrayList<>();
-  private final List<DataSegment> largeSegments = new ArrayList<>();
-  private final List<DataSegment> largeSegments2 = new ArrayList<>();
-  private DataSegment smallSegment;
-  private DruidCluster secondCluster;
-  private ServerHolder activeServer;
-  private ServerHolder decommissioningServer1;
-  private ServerHolder decommissioningServer2;
-  private SegmentLoadQueueManager loadQueueManager;
-
-  private static final String DS_SMALL = "small_source";
+  private int serverId = 0;
```

Comment:
The new tests are easier to understand and looks good; however, I haven't verified parity between the new and old tests.

Reply:
There are some changes:
- new tests have been added
- a couple of redundant tests have been removed
- all behaviour that was being verified earlier is still being verified.